tests/api/reports-categories.php

```diff
@@ -147,6 +149,78 @@ public function test_get_reports_categories_param() {
 		$this->assertArrayHasKey( 'category', $category_report['_links'] );
 	}
 
+	/**
+	 * Test getting reports with sorting.
+	 *
+	 * @since 3.9.0
+	 */
+	public function test_get_reports_categories_sort() {
+		WC_Helper_Reports::reset_stats_dbs();
+		wp_set_current_user( $this->user );
+
+		// Populate all of the data.
+		$product = new WC_Product_Simple();
+		$product->set_name( 'Test Product' );
+		$product->set_regular_price( 25 );
+		$product->save();
+
+		$order = WC_Helper_Order::create_order( 1, $product );
+		$order->set_status( 'completed' );
+		$order->set_total( 100 ); // $25 x 4.
+		$order->save();
+
+		WC_Helper_Queue::run_all_pending();
+
+		// Populate all of the data.
+		$product = new WC_Product_Simple();
+		$product->set_name( 'Test Product 2' );
+		$product->set_regular_price( 100 );
+		$second_category_id = wp_create_category( 'Second Category' );
+		$product->set_category_ids( array( $second_category_id ) );
+		$product->save();
+
+		WC_Helper_Queue::run_all_pending();
+
+		$uncategorized_term = get_term_by( 'slug', 'uncategorized', 'product_cat' );
+		$params             = array(
+			'orderby'    => 'category',
+			'order'      => 'desc',
+			'categories' => $uncategorized_term->term_id . ',' . $second_category_id,
```

Comment:
I don't think including the `categories` param correctly covers the case we're looking for.

Resulting query with `categories`:

```sql
SELECT
				default_results.category_id AS category_id, items_sold, net_revenue, orders_count, products_count
			FROM
				( 
			SELECT
				wp_wc_category_lookup.category_tree_id as category_id, SUM(product_qty) as items_sold, SUM(product_net_revenue) AS net_revenue, COUNT(DISTINCT wp_wc_order_product_lookup.order_id) as orders_count, COUNT(DISTINCT wp_wc_order_product_lookup.product_id) as products_count
			FROM
				wp_wc_order_product_lookup
				JOIN wp_wc_order_stats ON wp_wc_order_product_lookup.order_id = wp_wc_order_stats.order_id LEFT JOIN wp_term_relationships ON wp_wc_order_product_lookup.product_id = wp_term_relationships.object_id LEFT JOIN wp_wc_category_lookup ON wp_term_relationships.term_taxonomy_id = wp_wc_category_lookup.category_id
			WHERE
				1=1
				AND wp_wc_category_lookup.category_tree_id IN (360,363) AND wp_wc_order_product_lookup.product_id IN (145347,145344,145341,145338,145335,145326,145323,145306,145307,145303,145304,145305) AND ( wp_wc_order_stats.status NOT IN ( 'wc-trash','wc-pending','wc-cancelled','wc-failed' ) ) AND wp_wc_category_lookup.category_tree_id IS NOT NULL AND wp_wc_order_product_lookup.date_created <= '2019-12-31 23:59:59' AND wp_wc_order_product_lookup.date_created >= '2019-01-01 00:00:00'
		
				GROUP BY
					wp_wc_category_lookup.category_tree_id
			 ) AS wp_wc_order_product_lookup
				RIGHT JOIN ( SELECT 360 AS category_id UNION SELECT 363 AS category_id ) AS default_results
					ON default_results.category_id = wp_wc_order_product_lookup.category_id JOIN wp_terms AS _terms ON default_results.category_id = _terms.term_id
			WHERE
				1=1
				
		
				ORDER BY
					_terms.name desc
```

Query without `categories` (but with category sort desc):

```sql
SELECT
				wp_wc_category_lookup.category_tree_id as category_id, SUM(product_qty) as items_sold, SUM(product_net_revenue) AS net_revenue, COUNT(DISTINCT wp_wc_order_product_lookup.order_id) as orders_count, COUNT(DISTINCT wp_wc_order_product_lookup.product_id) as products_count
			FROM
				wp_wc_order_product_lookup
				JOIN wp_wc_order_stats ON wp_wc_order_product_lookup.order_id = wp_wc_order_stats.order_id LEFT JOIN wp_term_relationships ON wp_wc_order_product_lookup.product_id = wp_term_relationships.object_id LEFT JOIN wp_wc_category_lookup ON wp_term_relationships.term_taxonomy_id = wp_wc_category_lookup.category_id JOIN wp_terms AS _terms ON wp_wc_category_lookup.category_tree_id = _terms.term_id
			WHERE
				1=1
				AND ( wp_wc_order_stats.status NOT IN ( 'wc-trash','wc-pending','wc-cancelled','wc-failed' ) ) AND wp_wc_category_lookup.category_tree_id IS NOT NULL AND wp_wc_order_product_lookup.date_created <= '2019-12-31 23:59:59' AND wp_wc_order_product_lookup.date_created >= '2019-01-01 00:00:00'
		
				GROUP BY
					wp_wc_category_lookup.category_tree_id
			
				ORDER BY
					_terms.name desc
```

The query with included categories doesn't seem to hit the `LEFT JOIN`/`JOIN` interaction we're trying to fix.

Reply:
@jeffstieler This is ready for another review.